SU2_CFD/include/solvers/CSolver.hpp

```diff
@@ -4352,6 +4354,34 @@ inline void CustomSourceResidual(CGeometry *geometry, CSolver **solver_container
   AD::EndNoSharedReading();
 }
 
+  /*!
+   * \brief Seed derivatives for all solution variables using a flat vector.
+   * \param[in] derivatives - Flat vector of derivative values (size nVar * nPoint)
+   */
+  void SetSolutionDerivatives(const vector<passivedouble>& derivatives, CVariable* nodes) {
```

Comment:
use su2matrix instead of vector passivedouble so you don't need to deal with the offset variable